imagej/imagej.py

```diff
@@ -616,6 +631,18 @@ def _format_value(self, value):
             final_value = '[' + temp_value + ']'
             return final_value
 
+        def get_window_manager(self):
+            """
+            Get the ImageJ1 window manager if legacy mode is enabled.  It may not work properly if in headless mode.
+            :return: WindowManager
+            """
+            if not ij.legacy_enabled:
+                raise ImportError("Your ImageJ installation does not support IJ1.  This function does not work.")
+            elif ij.ui().isheadless():
+                logging.warning("Headless mode not enabled.  The WindowManager may not function as expected.")
+            else:
+                return WindowManager
+
```

Comment:
In the following `window_to_xarray` method, let's make behavior use pure IJ2 if the legacy layer is not active.

Reply:
Change:
```python
            if ij.legacy().isActive():
                imp = self.image_plus(sync=sync)
                return self._ij.py.from_java(imp)
            else:
                dataset = self._ij.dataset().getActiveDataset()
                return self._ij.py_from_java(dataset)
```